asyncscope.md

```diff
@@ -607,6 +608,78 @@ class Camera {
 }
 ```
 
+## Recursively spawning work until completion
+Below are two ways you could recursively spawn work on a scope using `let_with_async_scope` or `counting_scope`. Both
+examples are perfectly valid ways to ensure that all new work spawned from previous work will complete.
+
+:::cmptable
+
+### let_with_async_scope
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler auto sch, ex::counting_scope& scope, tree& t) {
+  return ex::let_error(ex::schedule(sch) | then([sch, &]() noexcept {
+    if (t.left)
+      ex::spawn(scope, process(sch, scope, t.left.get()));
+    if (t.right)
+      ex::spawn(scope, process(sch, scope, t.right.get()));
+    do_stuff(t.data);
+  }), | ex::let_error([](auto& e) {
+    // log error
+    return just(-1);
+  });
```

Comment:
A few notes:
 - the `noexcept` on `then`'s callable will defeat your attempt to handle errors,
 - the `let_error` syntax is a bit off, and
 - the `just(-1)` completes with `set_value(int)`, which would be fine in general but makes the overall sender returned from `process()` one that you can't `spawn`.
```suggestion
  return ex::schedule(sch) | then([sch, &]() {
    if (t.left)
      ex::spawn(scope, process(sch, scope, t.left.get()));
    if (t.right)
      ex::spawn(scope, process(sch, scope, t.right.get()));
    do_stuff(t.data);
  }), | ex::let_error([](auto& e) {
    // log error
    return just();
  });
```